libs/constellation/src/constellation.cpp

```diff
@@ -526,24 +531,21 @@ bool Constellation::OnBringUpExternalNetwork(
   consensus_ = CreateConsensus(cfg_, stake_, beacon_setup_, beacon_, *chain_, *storage_,
                                external_identity_->identity());
 
-  if (cfg_.proof_of_stake)
-  {
-    consensus_->SetDefaultStartTime(params.start_time);
-    consensus_->SetMaxCabinetSize(params.cabinet_size);
+  consensus_->SetWhitelist(params.whitelist);
+  consensus_->SetDefaultStartTime(params.start_time);
+  consensus_->SetMaxCabinetSize(params.cabinet_size);
 
-    if (params.snapshot)
-    {
-      consensus_->Reset(*params.snapshot);
-    }
-    else
-    {
-      FETCH_LOG_INFO(LOGGING_NAME, "No snapshot to reset consensus with");
-    }
+  if (params.snapshot)
+  {
+    consensus_->Reset(*params.snapshot);
+  }
+  else
+  {
+    FETCH_LOG_INFO(LOGGING_NAME, "No snapshot to reset consensus with.");
   }
 
-  execution_manager_ = std::make_shared<ExecutionManager>(
-      cfg_.num_executors, cfg_.log2_num_lanes, storage_,
-      [this] { return std::make_shared<Executor>(storage_); }, tx_status_cache_);
+  // Update with genesis to trigger loading any saved state
+  consensus_->UpdateCurrentBlock(*chain_->CreateGenesisBlock());
```

Comment:
This doesn't seem right?

Reply:
This should be the heaviest block on the chain presumably?

Reply:
I think it is fine - you are allowed to update consensus with a current block that is not contigous in block number so while this looks weird I think it is safe. I did previously have a solution that updated it to the current block that we recovered to in constellation but I deemed it unneccessary code.